kernel/src/vm/vmar/mod.rs

```diff
@@ -183,6 +205,9 @@ pub(super) struct Vmar_ {
     vm_space: Arc<VmSpace>,
     /// The RSS counters.
     rss_counters: [PerCpuCounter; NUM_RSS_COUNTERS],
+
+    /// OQueue Producer to notify policies about page fault events
+    page_fault_oqueue_producer: Mutex<Box<dyn Producer<PageFaultOQueueMessage>>>,
```

Comment:
OQueueRef supports a direct produce call which does not require external synchronization. That might be faster and simpler. We can revisit the "excessive capabilities" issue later.

Reply:
nvm OQueueRef it is